khelo/lib/ui/flow/team/add_team_member/add_team_member_screen.dart

```diff
@@ -105,184 +95,141 @@ class AddTeamMemberScreen extends ConsumerWidget {
               ),
             )
           : ListView.separated(
-              separatorBuilder: (context, index) {
-                return const SizedBox(
-                  height: 16,
-                );
-              },
+              padding: context.mediaQueryPadding +
+                  const EdgeInsets.symmetric(vertical: 16),
               itemCount: state.searchedUsers.length,
               itemBuilder: (context, index) {
                 UserModel user = state.searchedUsers[index];
-                return _userProfileCell(context, notifier, state, user);
+                return UserDetailCell(
+                  user: user,
+                  onTap: () => UserDetailSheet.show(context, user),
+                  trailing: OnTapScale(
+                    onTap: widget.team.players?.contains(user) != true &&
+                            !state.selectedUsers.contains(user)
+                        ? () async {
+                            if (user.phone != null) {
+                              final res = await VerifyAddTeamMemberDialog.show(
+                                  context,
+                                  phoneNumber: user.phone!
+                                      .substring(user.phone!.length - 5));
+                              if (res != null && res && context.mounted) {
+                                notifier.selectUser(user);
+                              }
+                            }
+                          }
+                        : null,
+                    child: Container(
+                      padding: const EdgeInsets.symmetric(
+                          horizontal: 16, vertical: 8),
+                      decoration: BoxDecoration(
+                          color: context.colorScheme.containerLow,
+                          borderRadius: BorderRadius.circular(20)),
+                      child: Text(
+                        widget.team.players?.contains(user) == true ||
+                                state.selectedUsers.contains(user)
+                            ? context.l10n.add_team_member_added_text
+                            : context.l10n.common_add_title.toUpperCase(),
+                        style: AppTextStyle.body2
+                            .copyWith(color: context.colorScheme.textDisabled),
+                      ),
+                    ),
+                  ),
+                );
               },
+              separatorBuilder: (context, index) => const SizedBox(height: 16),
```

Comment:
can we move it above itemBuilder?

Reply:
No, because seperater builder perform after build item so according to me this separatorBuilder is right position.